.circleci/config.yml

```diff
@@ -10,35 +10,62 @@ jobs:
     steps:
       - checkout
 
-      #TODO: add caching once demos share the same requirements file and can live in the same venv
-      #TODO: remove hard code in for-loops
+      # Restore cache
+      - run: &set-bash-env-template
+          name: Set bash environment
+          command: |
+            # Storing values shared between run calls
+            # Note: TEMP_FILE value is hardcoded in restore_cache and save_cache's checksum
+            echo 'export TEMP_FILE="temp_requirements_summary.txt"' >> $BASH_ENV
+            echo 'export IGNORED_DIRS="envs|tests|circuit-fault-diagnosis|factoring"' >> $BASH_ENV
+            . $BASH_ENV
+      - run: &set-requirements-summary-template
+          name: Create requirements summary
+          command: |
+            shopt -s extglob
+
+            # Clear file contents if it exists
+            > "$TEMP_FILE"
+
+            # Concatenate demo requirements.txt files into a temp file
+            for demo in !($IGNORED_DIRS); do
+              if [ -d "$demo" ]; then
+                echo "$demo" >> "$TEMP_FILE"
+                cat "$demo/requirements.txt" >> "$TEMP_FILE"
+                echo "" >> "$TEMP_FILE"
```

Comment:
Do you need the empty line in the end? If yes, at least you can lose the quotes here.

Reply:
The empty line is a nice-to-have for future debugging (i.e. a human needs to look through this temp file). I have removed the quotations